### PR TITLE
Update Rpc Errors

### PR DESCRIPTION
**Pull request by jelilat**

### CODE REVIEW

File: jsonrpc/server.go [hunk 1]
```diff
@@ -254,19 +254,13 @@ func (s *Server) handleRequest(req *request) (*response, error) {
 
 	calledMethod, found := s.methods[req.Method]
 	if !found {
-		res.Error = &Error{
-			Code:    MethodNotFound,
-			Message: "method not found",
-		}
+		res.Error = rpcErr(MethodNotFound, "method not found")
```
Comment (omerfirmak) [inline jsonrpc/server.go:257]: I think we can just pass `nil` as `data` here, right? It looks redundant.

**Review by omerfirmak — APPROVED** [review]

**Review by wojciechos — APPROVED** [review]

**Review by IronGauntlets — CHANGES_REQUESTED** [review]

File: jsonrpc/server.go (hunk 1 above)
Comment (omerfirmak) [reply jsonrpc/server.go]: should be `nil` not empty string

Comment (jelilat) [reply jsonrpc/server.go]: Done

File: jsonrpc/server.go [hunk 2]
```diff
@@ -56,6 +41,21 @@ type Error struct {
 	Data    any    `json:"data,omitempty"`
 }
 
+func rpcErr(code int, data interface{}) *Error {
```
Comment (omerfirmak) [inline jsonrpc/server.go:44]:
```suggestion
func rpcErr(code int, data any) *Error {
```

**Review by IronGauntlets — APPROVED** [review]

**Pull request merged**